core/include/vc/core/types/Volume.hpp

```diff
@@ -217,6 +219,8 @@ class Volume : public DiskBasedObjectBaseClass,
     bool cacheSlices_{true};
     /** Slice cache */
     mutable SliceCache::Pointer cache_{DefaultCache::New(DEFAULT_CAPACITY)};
+    /** Cache mutex for thread-safe access */
+    mutable std::mutex cacheMutex_;
```

Comment:
`shared_mutex` would still lead to the simultaneous read error, so switched to `mutex`.